### ISSUE TITLE
Metrics for GT papers

### ISSUE THREAD

**Issue by Charles-Fieseler-Vienna** [Zimmer-lab/barlow_track#48]
I think we should use the metrics in this paper, at least for the gt:
https://academic.oup.com/bioinformatics/article/30/11/1609/283435#393820269

**Comment by Charles-Fieseler-Vienna** [comment]
Paper from Harvard (tracker):
https://www-nature-com.uaccess.univie.ac.at/articles/s41592-023-02096-3

**Comment by Charles-Fieseler-Vienna** [comment]
Look at match_segmentation_and_tracks_using_config in wbfm repo... sorry it's messy!

**Comment by Charles-Fieseler-Vienna** [comment]
Here are all the pretrained networks:
/lisc/scratch/neurobiology/zimmer/wbfm/TrainedBarlow

**Comment by Charles-Fieseler-Vienna** [comment]
Histograms of accuracy across all hyperparameters, then do the same across datasets:
https://arxiv.org/pdf/1807.04587

**Comment by Charles-Fieseler-Vienna** [comment]
Path to output of the MIT tracking:
/lisc/scratch/neurobiology/zimmer/fieseler/wbfm_projects_future/flavell_data/projects/2025_07_01/3-tracking/barlow_tracker/df_barlow_tracks.h5

Raw data for MIT:
/lisc/scratch/neurobiology/zimmer/fieseler/wbfm_projects_future/flavell_data/projects/2025_07_01/nwb/flavell_data.nwb

**Comment by Charles-Fieseler-Vienna** [comment]
My implementation:
wbfm/utils/performance/comparing_ground_truth.py

Competitor paper:
https://www.biorxiv.org/content/10.1101/2025.07.23.666425v1.full.pdf

**Comment by Charles-Fieseler-Vienna** [comment]
More details about using my functions:
1. Match column names (only works well if the tracking is decent); I use rename_columns_using_matching() from wbfm/utils/neuron_matching/utils_candidate_matches.py
2. Actually calculate accuracy; I used calculate_accuracy_from_dataframes() from wbfm/utils/performance/comparing_ground_truth.py

**Comment by Charles-Fieseler-Vienna** [comment]
My (messy) jupyter notebook export that uses the second function; the column names are the same:

wbfm/notebooks/paper/supplement/fig_S1_tracking_accuracy.py

**Comment by Charles-Fieseler-Vienna** [comment]
To load the right dataframes, do something like this (but for the .nwb project as gt):

```
from wbfm.utils.projects.finished_project_data import ProjectData

project_path = '/scratch/neurobiology/zimmer/fieseler/wbfm_projects/project_pytest/project_config.yaml'


project_data = ProjectData.load_final_project_data(project_path, allow_hybrid_loading=True)
print(project_data)
print(project_data.intermediate_global_tracks.head())  # Maybe the nwb files will only have this field, not the final tracks
print(project_data.final_tracks.head())
```

**Comment by siegfriedS98** [comment]
@Charles-Fieseler-Vienna when loading the project_config.yaml from a finished trial, only the x,y,z coordinates are loaded and not the raw_segmentation_id etc

**Comment by Charles-Fieseler-Vienna** [comment]
Try updating the wbfm repo
